src/Microsoft.ML.Data/Evaluators/MultiOutputRegressionEvaluator.cs

```diff
@@ -385,7 +385,8 @@ private static VersionInfo GetVersionInfo()
                 verWrittenCur: 0x00010001, // Initial
                 verReadableCur: 0x00010001,
                 verWeCanReadBack: 0x00010001,
-                loaderSignature: LoaderSignature);
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(MultiOutputRegressionPerInstanceEvaluator).Assembly.FullName);
```

Comment:
>typeof(MultiOutputRegressionPerInstanceEvaluator).Assembly.FullName [](start = 36, length = 67)

So, would it be at all possible for this to have, if left as default or `null`, for `Assembly.GetCallingAssembly` to be invoked, since every time this is constructed I *think* it should be in the same class and assembly as is calling this constructor?

Not sure if this is too dangerous with inlining.

Reply:
I did a small benchmark of using `Assembly.GetExecutingAssembly`, and it was considerably slower than using the `typeof` approach, which is why I opt'ed to go with the `typeof` approach.

I didn't consider the `Assembly.GetCallingAssembly`, but it could possibly work in the default cases, but I'm not sure if we should rely on it all over. My thinking was that it is best to be explicit.

I can make the change if you feel strongly about it.

Reply:
Well, I don't know, what would have been a change in a handful of files has instead changed to a change in a hundred plus files, because of this choice. So would it not be better to not do this?

---
In reply to: [219941373](https://github.com/dotnet/machinelearning/pull/970#discussion_r219941373) [](ancestors = 219941373)

Reply:
From the [msdn Remarks section](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.getcallingassembly)

> If the method that calls the GetCallingAssembly method is expanded inline by the just-in-time (JIT) compiler, or if its caller is expanded inline, the assembly that is returned by GetCallingAssembly may differ unexpectedly.

I think explicitly specifying the assembly name is probably the most correct thing to do here.